spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/AggregateChangeExecutor.java

```diff
@@ -110,12 +114,12 @@ private void execute(DbAction<?> action, JdbcAggregateChangeExecutionContext exe
 			} else {
 				throw new RuntimeException("unexpected action");
 			}
-		} catch (Exception e) {
+		} catch (RuntimeException e) {
```

Comment:
It seems to me that we should revisit our exception handling. `DbActionExecutionException` doesn't seem particularly useful to the final application whereas it might be useful for our own internals.

Reply:
@mp911de Yeah... But this is not a part of this issue. Maybe we can create another issue to revise the default exception to be thrown? Or what should we do?

Reply:
The problem is that we're throwing `DbActionExecutionException` and not a DAO exception, so I think that is indeed part of the issue. Looking at the code, we do not catch `DbActionExecutionException`. That exception was introduced as part of introducing better means for debugging (#382) while the net effect was that we changed the API experience. What we could do is add `DbActionExecutionException` as suppressed exception to the original one.

Paging @schauder.

Reply:
Something along the lines of:

```
catch(RuntimeException e) {
  e.addSuppressed(new DbActionExecutionException(action));
  throw e;
}
```

Reply:
I'm a little bit confused with adding suppressed exception here. We're not really suppressing `DbActionExecutionException` here. I see it confusing to the end user TBH. 

CC: @mp911de @schauder

Reply:
I didn't know `addSuppressed`. It is a little hackish, but seems fine to me. It's not worse then the other variants I had in mind.

Reply:
Done